sema/check_import_declaration.go

```diff
@@ -45,15 +45,55 @@ func (checker *Checker) VisitImportDeclaration(declaration *ast.ImportDeclaratio
 	})
 }
 
-func (checker *Checker) declareImportDeclaration(declaration *ast.ImportDeclaration) {
+func (checker *Checker) declareImportDeclaration(
+	declaration *ast.ImportDeclaration,
+	allImported map[Imported]struct{},
+) {
 	locationRange := ast.NewRange(
 		checker.memoryGauge,
 		declaration.LocationPos,
-		// TODO: improve
-		declaration.LocationPos,
+		declaration.EndPos,
+	)
+
+	var (
+		aliases     map[string]string
+		usedAliases map[string]struct{}
 	)
 
-	identifiers := declaration.Identifiers
+	var identifiers []ast.Identifier
+	if len(declaration.Imports) > 0 {
+		identifiers = make([]ast.Identifier, 0, len(declaration.Imports))
+
+		for _, imp := range declaration.Imports {
+			identifiers = append(identifiers, imp.Identifier)
+
+			// Ensure that an alias is only used once
+
+			aliasName := imp.Alias.Identifier
+			if aliasName != "" {
+				importName := imp.Identifier.Identifier
+
+				if aliases == nil {
+					aliases = make(map[string]string)
+				}
+				aliases[importName] = aliasName
+
+				if usedAliases == nil {
+					usedAliases = make(map[string]struct{})
+				}
+
+				if _, ok := usedAliases[aliasName]; ok {
+					checker.report(
+						&DuplicateImportAliasError{
+							Alias: imp.Alias,
+						},
+					)
+				} else {
+					usedAliases[aliasName] = struct{}{}
+				}
```

Comment:
Do we report errors for cases where there's an alias with the same name as the imported identifier (not a duplicate alias).
e.g:
```cadence
import Foo as Bar from 0x1
import Bar from 0x2
```

Reply:
I just checked and though this is handled, the single declaration variant `import Foo as Bar, Bar from 0x1` isn't handled. I'll add tests and fix it

Reply:
Fixed in [5a003ac](https://github.com/onflow/cadence/pull/4199/commits/5a003ac19096f012d1d94d01fe1f0baa9215cc0b). Problem was that the alias was applied to early